pkg/kubelet/kubelet_network.go

```diff
@@ -89,6 +90,120 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	return supported
 }
 
+func omitDuplicates(kl *Kubelet, pod *v1.Pod, combinedSearch []string) []string {
+	uniqueDomains := map[string]bool{}
+
+	for _, dnsDomain := range combinedSearch {
```

Comment:
This is tricky enough that it warrants a unit test.  If this is all I find, I'll approve without it, and you can follow-up